java/libphonenumber/src/com/google/i18n/phonenumbers/PhoneNumberUtil.java

```diff
@@ -3337,7 +3345,7 @@ public MatchType isNumberMatch(PhoneNumber firstNumberIn, PhoneNumber secondNumb
     PhoneNumber secondNumber = copyCoreFieldsOnly(secondNumberIn);
     // Early exit if both had extensions and these are different.
     if (firstNumber.hasExtension() && secondNumber.hasExtension()
-        && !firstNumber.getExtension().equals(secondNumber.getExtension())) {
+        && !extensionMatches(firstNumber.getExtension(), secondNumber.getExtension())) {
```

Comment:
The Javadoc above should mention what happens with differences in control-characters (comma & semicolon) in the extensions.

Before addressing this, let's figure out the semantics. For that, please see my comment on testIsNumberMatchMatches.